### ISSUE TITLE
IsGroupAsSemigroup fails on some semigroups

### ISSUE THREAD

**Issue by saliola** [semigroups/Semigroups#875]
The [first example in the documentation](https://docs.gap-system.org/pkg/semigroups/doc/chap12_mj.html#X852F29E8795FA489) for `IsGroupAsSemigroup` does not work.

Here is what the documentation says should happen:
```
gap> S := Semigroup(Transformation([2, 4, 5, 3, 7, 8, 6, 9, 1]),
>                   Transformation([3, 5, 6, 7, 8, 1, 9, 2, 4]));;
gap> IsGroupAsSemigroup(S);
true
```
Here is what happens:
```
gap> S := Semigroup(Transformation([2, 4, 5, 3, 7, 8, 6, 9, 1]),
>                   Transformation([3, 5, 6, 7, 8, 1, 9, 2, 4]));;
gap> IsGroupAsSemigroup(S);
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 2nd choice method found for `IsGroupAsSemigroup' on 1 arguments at /.../gap/lib/methsel2.g:249 called from
<function "HANDLE_METHOD_NOT_FOUND">( <arguments> )
 called from read-eval loop at *stdin*:12
type 'quit;' to quit to outer loop
brk>
```

Same thing happens on the main branch as well as the stable-5.0 branch.

**Comment by saliola** [comment]
Wait, is this just because I forgot to load the semigroups package?!  😬

**Comment by saliola** [comment]
OK, I tracked down the problem. This is not a bug in Semigroups, rather an inconsistency with the documentation that confused me for a long time.

I am trying to use this from within SageMath, and according to the [first line under Installation](https://github.com/semigroups/Semigroups#:~:text=It%20is%20assumed%20that%20you%20have%20a%20working%20copy%20of%20GAP%20with%20version%20number%204.10.0%20or%20higher.) on the repo's landing page, I only need version GAP-4.10.0 or higher, which is true of the version of SageMath I'm using. However, this is not true for stable-5.0 and later, one actually needs at least GAP-4.12.

So maybe this issue can be changed to "please update the README to correct the minimal version of dependencies".

**Comment by james-d-mitchell** [comment]
Thanks for the report @saliola, we will update the README today.

**Comment by james-d-mitchell** [comment]
Just to check @saliola was this resolved by #876?

**Comment by saliola** [comment]
Yes, it is. Thank you.
